internal/mode/static/state/store_test.go

```diff
@@ -0,0 +1,52 @@
+package state
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	discoveryV1 "k8s.io/api/discovery/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	v1 "sigs.k8s.io/gateway-api/apis/v1"
+)
+
+func TestSetChangeType(t *testing.T) {
+	ctu := newChangeTrackingUpdater(nil, nil, nil)
+
+	tests := []struct {
+		obj     client.Object
+		name    string
+		exp     ChangeType
+		changed bool
+	}{
+		{
+			name: "no change",
+			exp:  NoChange,
+		},
+		{
+			name:    "endpoint object",
+			obj:     &discoveryV1.EndpointSlice{},
+			changed: true,
+			exp:     EndpointsOnly,
+		},
+		{
+			name:    "non-endpoint object",
+			obj:     &v1.HTTPRoute{},
+			changed: true,
+			exp:     ClusterStateChange,
+		},
+		{
+			name:    "changeType was previously set to ClusterStateChange",
```

Comment:
Can we add a comment that states the order matters for the test cases?

Reply:
related to this:

```
	for _, test := range tests {
		t.Run(test.name, func(t *testing.T) {
			g := NewWithT(t)
			ctu.setChangeType(test.obj, test.changed)
			g.Expect(ctu.changeType).To(Equal(test.exp))
		})
	}
```

is there are chance the order will not be consistent when running the tests, because of t.Run of each test case

Reply:
The outer loop should ensure that the order remains the same, right?

Reply:
It should. The test is simple enough that we could also just write it without the cases.